### PR TITLE
initial template for new PRs

### PR DESCRIPTION
**Pull request by ma2bd**
## Motivation

Standardize github Pull Requests and encourage best practices.

## Proposal

This template

## Test Plan

Copy pasted the template here. Looks like to need to land first on the main branch to be picked by GH.

## Links

* https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository

## Release Plan

- [x] All good!
- [ ] Need to bump the major/minor version number in the next release of the crates.
- [ ] Need to update the developer manual.
- [ ] This PR is adding Cargo features.
- [ ] This PR is removing or renaming existing Cargo features.
- [ ] Release is blocked and/or tracked by other issues (see links below)

## Reviewer Checklist

- [ ] The title and the summary of the PR are short and descriptive.
- [ ] The proposed solution achieves the goals stated in the PR.
- [ ] The test plan is reproducible and provides sufficient coverage.
- [ ] The release plan is adequate.
- [ ] The commits correspond to distinct logical changes.
- [ ] The code follows the [coding guidelines](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md).
- [ ] The proposed changes look correct.
- [ ] The CI is passing.
- [ ] All of the above.

### CODE REVIEW

**Review by afck — APPROVED** [review]

**Review by MathieuDutSik — APPROVED** [review]

**Comment by MathieuDutSik** [comment]
Maybe a section `Next steps` or similar could be proposed as an optionality.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+Please create issues to track future improvements.
+-->
+
+## Reviewer Checklist
+
+- [ ] The title and the summary of the PR are well-written.
+- [ ] The proposed solution makes sense.
+- [ ] The test plan and the release plan are convincing.
+- [ ] The commits are easy to review.
+- [ ] The code follows the [coding guidelines](https://github.com/ma2bd/linera-protocol/CONTRIBUTING.md).
```
Comment (andresilva91) [inline .github/pull_request_template.md:38]:
```suggestion
- [ ] The code follows the [coding guidelines](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md).
```

Comment (ma2bd) [reply .github/pull_request_template.md:38]: oops my bad

**Comment by ma2bd** [comment]
> I think some of these requirements are subjective, I would rephrase[...]

There will always be room for interpretation but happy to make some of the requirements more precise.
